rpc/eth_api.go

```diff
@@ -90,9 +90,20 @@ func (e *PublicEthAPI) Syncing() (interface{}, error) {
 	}, nil
 }
 
-// Coinbase returns this node's coinbase address. Not used in Ethermint.
-func (e *PublicEthAPI) Coinbase() (addr common.Address) {
-	return
+// Coinbase is the address that staking rewards will be send to (alias for Etherbase).
+func (e *PublicEthAPI) Coinbase() (common.Address, error) {
+	height, err := e.BlockNumber()
+	if err != nil {
+		return common.Address{}, err
+	}
+	blockNumber := int64(height)
+
+	block, err := e.cliCtx.Client.Block(&blockNumber)
+	if err != nil {
+		return common.Address{}, err
+	}
+
+	return common.BytesToAddress(block.Block.ProposerAddress.Bytes()), nil
```

Comment:
this looks like it returns the coinbase address on the current block, not the coinbase address of this node? 
https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_coinbase

not sure if this is functionally correct, what is this used for?

Reply:
I updated it to return the validator address of the node 👍

Reply:
nice :D